src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java

```diff
@@ -20,17 +18,17 @@
 //@RequestMapping("/films")
 public class FilmController {
     private final FilmService filmService;
-    private final UserService userService;
     @Autowired
-    public FilmController (FilmService filmService, UserService userService) {
+    public FilmController (FilmService filmService) {
         this.filmService = filmService;
-        this.userService = userService;
     }
 
+
     @PostMapping("/films")
-    public Film add(@Valid @RequestBody final Film film) throws FilmException, ValidationException { //добавление фильма
-        filmService.add(film);
+    public Film create(@Valid @RequestBody final Film film) throws FilmException, ValidationException, NotFoundException { //добавление фильма
```

Comment:
Если не передать MPA, то падает с NullPointerException
{
	"name":"FilmName",
	"description":"descriptionItem",
	"duration": 22,
	"releaseDate": "2012-04-23T18:25:43.511Z"
}

java.lang.NullPointerException: Cannot invoke "ru.yandex.practicum.filmorate.model.Mpa.getId()" because the return value of "ru.yandex.practicum.filmorate.model.Film.getMpa()" is null
	at ru.yandex.practicum.filmorate.storage.db.FilmDbStorage.lambda$save$1(FilmDbStorage.java:120) ~[classes/:na]
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:649) ~[spring-jdbc-5.3.22.jar:5.3.22]
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:991) ~[spring-jdbc-5.3.22.jar:5.3.22]
	at ru.yandex.practicum.filmorate.storage.db.FilmDbStorage.save(FilmDbStorage.java:113) ~[classes/:na]
	at ru.yandex.practicum.filmorate.service.FilmService.save(FilmService.java:94) ~[classes/:na]
	at ru.yandex.practicum.filmorate.controller.FilmController.create(FilmController.java:29)

Но пользователь получает корректное сообщение об ошибке, что неверные входные данные. Но это должен быть BadRequest, а не Internal Server Error

Reply:
Исправил. Дополнил валидацию.